operator/pkg/apis/istio/v1alpha1/validation/validation.go

```diff
@@ -86,11 +89,42 @@ func deprecatedSettingsMessage(values *valuesv1alpha1.Values) string {
 		{"pilot.ingress", "meshConfig.ingressService, meshConfig.ingressControllerMode, and meshConfig.ingressClass", nil},
 		{"global.mtls.enabled", "the PeerAuthentication resource", nil},
 		{"global.mtls.auto", "meshConfig.enableAutoMtls", nil},
+		{"telemetry.v1.enabled", mixerDeprecatedMessage, false},
+		{"global.disablePolicyChecks", mixerDeprecatedMessage, false},
+		{"pilot.policy.enabled", mixerDeprecatedMessage, false},
 	}
-	for _, d := range deprecations {
+	iopDeprecations := []deprecatedSettings{
+		{"components.telemetry.enabled", mixerDeprecatedMessage, false},
+		{"components.policy.enabled", mixerDeprecatedMessage, false},
+	}
+	for _, d := range valuesDeprecations {
 		v, f, _ := tpath.GetFromStructPath(values, firstCharsToUpper(d.old))
-		if f && v != d.def {
-			messages = append(messages, fmt.Sprintf("! %s is deprecated; use %s instead", d.old, d.new))
+		if f {
+			switch v.(type) {
+			case *gogotypes.BoolValue:
+				val, ok := v.(*gogotypes.BoolValue)
+				if ok {
+					v = val.Value
+				}
+			}
+			if v != d.def {
+				messages = append(messages, fmt.Sprintf("! %s is deprecated; use %s instead", d.old, d.new))
+			}
+		}
+	}
+	for _, d := range iopDeprecations {
```

Comment:
I have a similar PR that also addresses some issues with maps. Do you think we should do one or the other or some combination of these approaches?

Reply:
Can you link your PR?

Reply:
oops sorry I thought I did https://github.com/istio/istio/pull/24567

Reply:
yep I think what you are doing is more general, I can rebase my change on top of yours then